feature/src/adm.c

```diff
@@ -48,6 +53,228 @@ static char *init_dwt_band(adm_dwt_band_t *band, char *data_top, size_t buf_sz_o
     return data_top;
 }
 
+#if ADM_OPT_ENABLE
+static char *init_dwt_band_hvd(adm_dwt_band_t *band, char *data_top, size_t buf_sz_one)
```

Comment:
type of `data_top` should be a `float*`. Every call of this function, the data starts as a `float*` get casted to a a `char*` and in the function body gets casted back to a `float*`